csv-validator-core/src/main/scala/uk/gov/nationalarchives/csv/validator/api/CsvValidator.scala

```diff
@@ -74,53 +64,44 @@ trait CsvValidator extends SchemaParser {
    */
   def validate(csvFile: TextFile, csvSchema: Schema, progress: Option[ProgressCallback]): MetaDataValidation[Any] = {
    
-    val integrationValidation: MetaDataValidation[Any] = integrityCheckValidation(csvFile, csvSchema).getOrElse(true.successNel[FailMessage])
+    val integrationValidation: MetaDataValidation[Any] = integrityCheckValidation2(csvFile, csvSchema).getOrElse(true.successNel[FailMessage])
 
     val metadataValidation:MetaDataValidation[Any] = withReader(csvFile) {
       reader =>
         validateKnownRows(reader, csvSchema, progress.map(p => ProgressFor(countRows(csvFile), p)))
     }
-    //TODO Combine in a better depending in the strategie FailFast or not
+    //TODO Combine in a better depending in the strategy FailFast or not
     List(integrationValidation, metadataValidation).sequence[MetaDataValidation, Any]
   }
 
-  /**
-   * Check if all the file under the content folder are listed in the metadata file listing
-   * @param csvFile the CSV File
-   * @param csvSchema the CSV schema
-   * @return Option[MetaDataValidation] defined only the filename column is defined, and if so return the according metadata
-   */
-  def integrityCheckValidation(csvFile: TextFile, csvSchema: Schema): Option[MetaDataValidation[Any]] = {
-    integrityCheckFilenameColumn.map {filenameColumn =>
-      val columnDedinitionIds = csvSchema.columnDefinitions.map(_.id)
-      if (columnDedinitionIds.exists(_ == filenameColumn)) {
-
-        val filenameColumnIndex = columnDedinitionIds.indexOf(filenameColumn)
-        val filnameColumnRules = csvSchema.columnDefinitions.apply(filenameColumnIndex)
+  def integrityCheckValidation2(csvFile: TextFile, csvSchema: Schema): Option[MetaDataValidation[Any]] = {
```

Comment:
Should the integrityCheck not be done in parallel to the CSV validation? Otherwise I fear you will have a long wait of one for the other. Also it seems the integrity check could itself be parallelised.